extension/js/content_scripts/webmail/gmail_element_replacer.ts

```diff
@@ -130,28 +132,23 @@ export class GmailElementReplacer implements WebmailElementReplacer {
     const visibleReplyBtns = $('td.acX:visible');
     if (visibleReplyBtns.not('.replaced, .inserted').length) { // last reply button in convo gets replaced
       const convoReplyBtnsToReplace = visibleReplyBtns.not('.replaced, .inserted');
-      const hasVisibleReplacements = visibleReplyBtns.filter('.replaced, .inserted').length > 0;
       const convoReplyBtnsToReplaceArr = convoReplyBtnsToReplace.get();
-      if (!hasVisibleReplacements && convoReplyBtnsToReplaceArr.length) {
-        // only replace with FlowCrypt reply button if does not have any buttons replaced yet, and only replace the last one
-        const lastReplyBtn = $(convoReplyBtnsToReplaceArr.pop()!);
-        $(lastReplyBtn).addClass('inserted');
-        const element = $(this.factory.btnReply()).insertBefore($(lastReplyBtn).children().last());  // xss-safe-factory
-        if (isEncrypted) {
-          element.click(Ui.event.prevent('double', () => this.setReplyBoxEditable().catch(Catch.reportErr)));
-        } else {
-          element.click(Ui.event.prevent('double', Catch.try(() => this.replaceStandardReplyBox(true, true)))); // xss-safe-factory
-        }
-      }
-      if (isEncrypted) {
-        // hide original gmail reply buttons, except for the last one, but ONLY IF this is an encrypted conversation
-        // the reason is that currently encrypted replies only support replying to last message. If some other reply button
-        // was clicked, and FlowCrypt replaced the reply box with an encrypted one mid-message, it would create a false imagination
-        // for the user that they are replying mid-convo, when in fact it would end up replying to the last message anyway
-        // todo - revisit this (and potentially remove) when we support encrypted replies mid-convo
-        for (const replyBtn of convoReplyBtnsToReplaceArr) { // all other non-last reply buttons to be hidden
-          $(replyBtn).addClass('replaced').text(''); // hide all except last
-        }
+      // only replace with FlowCrypt reply button if does not have any buttons replaced yet, and only replace the last one
+      for (const elem of convoReplyBtnsToReplaceArr) {
+        $(elem).addClass('inserted');
+        const element = $(this.factory.btnReply()).insertBefore($(elem).children().last());  // xss-safe-factory
+        const messageContainer = $(elem.closest('.h7') as HTMLElement);
```

Comment:
I imagine this could be a memory leak, because `messageContainer` will be used at unknown time in the future inside `element.click`, and the content scripts are similar to SPAs (will never have a chance to de-allocate memory in this case).

We actually already have a few memory leaks in the content script that customers were complaining about.

Is it possible to compute/ find `messageContainer` inside the handler, based on which element was clicked?

Reply:
yes, it is possible, will do